pkg/controller/restore_controller_test.go

```diff
@@ -922,6 +922,62 @@ func TestMostRecentCompletedBackup(t *testing.T) {
 	assert.Equal(t, expected, mostRecentCompletedBackup(backups))
 }
 
+// OADP Carry: Test that restore that has status inProgress on reconcile is changed to failed if velero has memory of it failing.
```

Comment:
@sseago from a rebase perspective is this ok? would prefer another file or something different?

Reply:
rebase perspective is ok from [slack](https://redhat-internal.slack.com/archives/CHD1CSNAK/p1719251370499649?thread_ts=1718980526.954259&cid=CHD1CSNAK).

Adding cases where err is expected from Reconcile when patch fails again.

Reply:
@weshayutin It's a separate test func, so it should be fine in the same file.